namui-prebuilt/src/event_trap.rs

```diff
@@ -0,0 +1,61 @@
+use namui::prelude::*;
+
+pub fn event_trap(content: RenderingTree) -> RenderingTree {
+    content.attach_event(move |builder| {
+        builder
+            .on_mouse_move_in(|event| {
+                namui::log!("on_mouse_move_in");
+                event.stop_propagation()
+            })
+            .on_mouse_move_out(|event| {
+                namui::log!("on_mouse_move_out");
+                event.stop_propagation()
+            })
+            .on_mouse_down_in(|event| {
+                namui::log!("on_mouse_down_in");
+                event.stop_propagation()
+            })
+            .on_mouse_down_out(|event| {
+                namui::log!("on_mouse_down_out");
+                event.stop_propagation()
+            })
+            .on_mouse_up_in(|event| {
+                namui::log!("on_mouse_up_in");
+                event.stop_propagation()
+            })
+            .on_mouse_up_out(|event| {
+                namui::log!("on_mouse_up_out");
+                event.stop_propagation()
+            })
+            .on_wheel(|event| {
+                namui::log!("on_wheel");
+                event.stop_propagation()
+            });
+        // below don't support stop_propagation
+        // .on_key_down(|event| event.stop_propagation())
+        // .on_key_up(|event| event.stop_propagation())
+    })
+}
+
+pub fn event_trap_mouse(content: RenderingTree) -> RenderingTree {
+    content.attach_event(|builder| {
+        builder
+            .on_mouse_down_in(|event| {
+                event.stop_propagation();
+            })
+            .on_mouse_move_in(|event| {
+                event.stop_propagation();
+            })
+            .on_wheel(|event| {
+                let xy = event.namui_context.get_rendering_tree_xy(event.target);
+                if let Some(xy) = xy {
+                    if let Some(bounding_box) = event.target.get_bounding_box() {
+                        let bounding_box = bounding_box + xy;
+                        if bounding_box.is_xy_inside(system::mouse::position()) {
+                            event.stop_propagation();
+                        }
+                    }
+                }
+            });
+    })
+}
```

Comment:
It passes mouse down event. Then why is this event_trap_**mouse**?

Reply:
you mean mouse up?

Reply:
I just push this commit [Add mouse up](https://github.com/NamseEnt/namseent/pull/425/commits/b31aa9af52c89dc6b169fb971b91bb3329db9fb4)

Reply:
Oh, sorry. I thought it was `on_mouse_move_in` and `on_mouse_move_out`.
Is there no need for a mouse up?